src/replication.c

```diff
@@ -1940,6 +1966,7 @@ replicate(ISTGT_LU_DISK *spec, ISTGT_LU_CMD_Ptr cmd, uint64_t offset, uint64_t n
 
 			MTX_LOCK(&spec->rq_mtx);
 			TAILQ_REMOVE(&spec->rcommon_waitq, rcomm_cmd, wait_cmd_next);
+			put_to_mempool(&spec->rcommon_deadlist, rcomm_cmd);
```

Comment:
Since mempool is lockless multithread safe data structure, please move  put_to_mempool call after releasing lock.